### PR TITLE
Don't include XML when serializing AFF

### PR DESCRIPTION
**Pull request by mbtools**
Preparation for #5126

- Avoids output of metadata XML when serialized object contains JSON (ABAP file format).
- Contains new and updated unit tests
- Removes obsolete exceptions

### CODE REVIEW

File: src/objects/core/zcl_abapgit_objects_files.clas.abap [hunk 1]
```diff
@@ -88,6 +80,9 @@ CLASS zcl_abapgit_objects_files DEFINITION
     METHODS get_file_pattern
       RETURNING
         VALUE(rv_pattern) TYPE string .
+    METHODS is_aff
```
Comment (larshp) [inline src/objects/core/zcl_abapgit_objects_files.clas.abap:83]: rename to `is_json` ?

Comment (flaiker) [reply src/objects/core/zcl_abapgit_objects_files.clas.abap:83]: is_aff seems to be more precise as looking just at the method name is_json might also include my_file.w3mi.data.json

Comment (mbtools) [reply src/objects/core/zcl_abapgit_objects_files.clas.abap:83]: the method doesn't look into the content of the `json` to determine if it's indeed `aff`. how about `is_json_metadata`?

Comment (larshp) [reply src/objects/core/zcl_abapgit_objects_files.clas.abap:83]: yea

Comment (mbtools) [reply src/objects/core/zcl_abapgit_objects_files.clas.abap]: done

File: src/objects/core/zcl_abapgit_objects_files.clas.abap [hunk 2]
```diff
@@ -248,6 +243,26 @@ CLASS zcl_abapgit_objects_files IMPLEMENTATION.
...
+  METHOD is_json_metadata.
+
+    DATA lv_regex TYPE string.
+
+    FIELD-SYMBOLS <ls_file> LIKE LINE OF mt_files.
+
+    lv_regex = '^[^\.]+\.' && to_lower( ms_item-obj_type ) && '\.json$'.
+
+    LOOP AT mt_files ASSIGNING <ls_file>.
+      FIND REGEX lv_regex IN  <ls_file>-filename.
```
Comment (larshp) [inline src/objects/core/zcl_abapgit_objects_files.clas.abap:255]: the regex seems overkill? its just `CP |*.{ to_lower( ms_item-obj_type ) }.json|` ?

**Comment by larshp** [comment]
ok

but I see it as a temporary workaround, json should be abstracted on the same level as xml

**Comment by larshp** [comment]
fyi @albertmink

**Pull request merged**